demo/src/TestArea.js

```diff
@@ -1,6 +1,26 @@
 import React, { Component } from 'react'
 
 class TestArea extends Component {
+  constructor (props) {
+    super(props)
+
+    this.timeout = null
+    this._onKeyUp = this._onKeyUp.bind(this)
+  }
+
+  _onKeyUp() {
+    if (this.timeout) {
+      clearTimeout(this.timeout)
+    }
+
+    this.props.startTyping()
+
+    // stop typing after 1 second of inactivity
+    this.timeout = setTimeout(() => {
+      this.props.stopTyping()
+    }, 1000)
```

Comment:
a bit of a nit since this is just the example but, there are no semi-colons on any of these lines